src/goose.rs

```diff
@@ -1541,6 +1541,23 @@ pub fn get_base_url(
     }
 }
 
+// pub function: for<'r> fn(&'r GooseUser) -> Pin<Box<dyn Future<Output = ()> + Send + 'r>>,
+pub trait GooseTaskCallback<'a> {
+    type Output: std::future::Future<Output = ()> + Send + 'a;
+    fn call(self, arg: &'a GooseUser) -> Self::Output;
+}
+
+impl<'a, Fut: 'a, F> GooseTaskCallback<'a> for F
+where
+    F: FnOnce(&'a GooseUser) -> Fut,
+    Fut: std::future::Future<Output = ()> + Send,
```

Comment:
> Would this allow us to pass in a closure as well?

Yes. Closures implement FnOnce, which is what this blanket impl covers.

Reply:
... actually I tried it and it turns out it doesn't get the lifetimes right. That's sad.

Reply:
@alsuren Any ideas on how we could make this work with closures? That's the big inspiration for any further changes to how wrapping tasks works, to make Goose far more useful as a library.

Reply:
I'm afraid I've not had the head space for this after work, so I've not looked into it.

Next things I would try would probably be:
* See if there is a way to annotate the lifetimes on the closure as you create it
* Take a step back to the previous approach (which managed to use closures as part of the workflow just fine, but couldn't accept them as a macro argument because we only accept ident as input) and see if there is a way to accept a closure as a macro argument. (Prototype this without using macros first maybe?)